lib/phoenix/endpoint/attempt_code_reload_plug.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Phoenix.Endpoint.AttemptCodeReloadPlug do
+  @moduledoc ~S"""
+  Wraps an Endpoint, attempting to sync with Phoenix's code reloader if 
+  an exception is raising which indicates that we may be in the middle of a reload.
+
+  We detect this by looking at the raised exception and seeing if it indicates
+  that the endpoint is not defined. This indicates that the code reloader may be 
+  mid way through a compile, and that we should attempt to retry the request
+  after the compile has completed. This is also why this must be implemented in
+  a separate module (one that is not recompiled in a typical code reload cycle),
+  since otherwise it may be the case that the endpoint itself is not defined.
```

Comment:
I wonder if we need this now?

Before we needed it because the handler would call the endpoint separately. But in this approach the handler no longer calls the endpoint, so it won't run into races.

In fact, the race we can run into is that we won't find the user handler (i.e. the UserSocket module), so we would need to wrap that instead.